src/common/evaluable.ts

```diff
@@ -55,10 +55,17 @@ export interface Evaluable {
    * Simplifies this Evaluable when possible.
    *
    * @param {Context} ctx context for the evaluation
-   * @param {string[]} ignoreKeys keys to be considered present even if their not present in the context
+   * @param {string[]} strictKeys keys to be considered present even if they are not present in the context
+   * @param {string[]} optionalKeys keys to be considered not present unless they are also in `strictKeys`;
+   *  when `strictKeys` is `undefined` and `optionalKeys` is an array, every key that is not in `optionalKeys`
+   *  is considered to be present and thus will be evaluated
```

Comment:
Reading this description carefully, it doesn't seem to reflect exact what is happening. Because we only check `optionalKeys` when the value is not present on the context.

Reply:
Done 245d7e99a7819e75c4d48d14697fc2d608bd5c51